src/main/java/io/vertx/core/shareddata/AsyncMap.java

```diff
@@ -134,4 +141,90 @@
    */
   void size(Handler<AsyncResult<Integer>> resultHandler);
 
+  /**
+   * Get the keys of the map, asynchronously.
+   * <p>
+   * Use this method with care as the map may contain a large number of keys,
+   * which may not fit entirely in memory of a single node.
+   * In this case, the invocation will result in an {@link OutOfMemoryError}.
+   * <p>
+   * When in doubt, prefer usage of {@link #keyStream()}.
```

Comment:
that's not really true since Hazelcast all the keys are loaded right ?

Reply:
Yes but the `xxxStream` methods have a warning that not all cluster manager support true streaming.

Reply:
I'm worried that Hazelcast does not provide streaming yet for this, so it would only be effective with Infinispan which is today a very minority of the users, so I would like for now to remove streaming and defer until we can support it in the general case, otherwise we are going to have people running into issues and we will have no solution for them.

Reply:
Not only Infinispan, Ignite too. Anyway I went back to Hazelcast doc and I believe we could do with queries and a `PagingPredicate`. Let me update the HZ cluster manager PR.

Reply:
@vietj as discussed offline, I talked with Hazelcast guys and PagingPredicates are not a good solution: they require a fullscan on every iteration.
So I went ahead and removed the streaming methods from the generic API as well as Ignite, Hazelcast and Zookeeper cluster managers.
In Infinispan I added a new interface for extra features. The streaming features are maintained there.